### PR TITLE
[DISCO-3567] Weather: return admin code or country name based on request

### PR DESCRIPTION
**Pull request by tiftran**
## References

JIRA: [DISCO-3567](https://mozilla-hub.atlassian.net/browse/DISCO-3567)

## Description
Since countries outside of the US and CA find it odd that locations are displayed as <City>, <Admin Code> rather than <City>, <Country>
Modifications are made such that depending on the city request for weather and the country of where the request is coming from. We change region_code to use either admin code or country name.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [ ] The PR title starts with the JIRA issue reference, format example `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|skip|warn)]` keywords are applied to the last commit message (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-3567]: https://mozilla-hub.atlassian.net/browse/DISCO-3567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/MC-1747)

### CODE REVIEW

**Review by ncloudioj — APPROVED** [review]
LGTM, thanks!

File: merino/providers/suggest/weather/backends/accuweather/backend.py [hunk 1]
```diff
@@ -501,7 +504,26 @@ def get_localized_city_name(
...
+    def get_region_for_weather_report(
+        location: AccuweatherLocation, weather_context: WeatherContext
+    ) -> str | None:
+        """Get region based on request country origin and the country of the requested city."""
+        geolocation = weather_context.geolocation
+        # we don't override country_name so it should tell us where the request came from
+        request_origin_country = geolocation.country_name
+        requested_country = geolocation.country
+
+        # None in the list because if we don't know the origin country of the request, use NA convention.
```
[the remark this answers is not in the document]
Comment (tiftran) [reply merino/providers/suggest/weather/backends/accuweather/backend.py:519]: edge case: ip does not map to a geolocation

File: merino/providers/suggest/weather/backends/accuweather/backend.py [hunk 2]
```diff
@@ -155,6 +155,9 @@ class AccuweatherLocation(BaseModel):
     # Unique Administrative Area ID for the Location.
     administrative_area_id: str
 
+    # Country name for the location
+    country_name: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (tiftran) [reply merino/providers/suggest/weather/backends/accuweather/backend.py:159]: ah yup, you're right

File: merino/providers/suggest/weather/backends/accuweather/backend.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (0c0w3) [reply merino/providers/suggest/weather/backends/accuweather/backend.py:519]: I can't speak to this conversation but wanted to say, if we don't know the origin of the request, we should return `country_name` here IMO. Showing "{city}, {country}" is always correct but "{city}, {region_code}" seems to be wrong for a lot (most?) countries.

[the remark this answers is not in the document]
Comment (tiftran) [reply merino/providers/suggest/weather/backends/accuweather/backend.py]: made changes, since yeah edge case should be tiny and while it looks funny for me locally cause ip doesn't map to anything to the test maxmind db, that's a me problem 🙃

**Review by 0c0w3 — APPROVED** [review]
The core `get_region_for_weather_report` change and tests lgtm, thanks!
